node/src/accountant/scanners.rs

```diff
@@ -32,6 +33,8 @@ use std::cell::RefCell;
 use std::rc::Rc;
 use std::sync::{Arc, Mutex};
 use std::time::SystemTime;
+use time::format_description::parse;
```

Comment:
This doesn't relate to imports :). I just needed a general place to put my comment...  :man_shrugging: 

Those three last methods of this trait extremely repeat through this file.
You will do right if you use a macro. To justify myself:

a) the methods are very simple, so they can be read easily even from a macro,

b) they really appear in each implementation without any difference 

c) the space (and so the attention) they take is just distraction; the reader should instead pay attention to more meaningful functions such as `begun_scan()` and `end_scan()`. 

I wrote the macro for you, though it's pretty simple. 
```

macro_rules! time_marking_methods {
    ($scan_type_variant: ident) => {
        fn scan_started_at(&self) -> Option<SystemTime> {
            self.common.initiated_at_opt
        }

        fn mark_as_started(&mut self, timestamp: SystemTime) {
            self.common.initiated_at_opt = Some(timestamp);
        }

        fn mark_as_ended(&mut self, logger: &Logger) {
            self.common.zero_timestamp_out_and_log(
                ScanType::$scan_type_variant,
                logger,
            );
        }
    }
} 
```